test/extensions/filters/http/cache/cache_filter_integration_test.cc

```diff
@@ -0,0 +1,84 @@
+#include "test/integration/http_protocol_integration.h"
+#include "test/test_common/simulated_time_system.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Cache {
+
+// TODO(toddmgreer): Expand integration test to include age header values,
+// expiration, range headers, HEAD requests, trailers, config customizations,
+// cache-control headers, and conditional header fields, as they are
+// implemented.
+
+class CacheIntegrationTest : public Event::TestUsingSimulatedTime,
+                             public HttpProtocolIntegrationTest {
+public:
+  void TearDown() override {
+    cleanupUpstreamAndDownstream();
+    HttpProtocolIntegrationTest::TearDown();
+  }
+
+  void initializeFilter(const std::string& config) {
+    config_helper_.addFilter(config);
+    initialize();
+    codec_client_ = makeHttpConnection(makeClientConnection((lookupPort("http"))));
+  }
+
+  const std::string default_config{R"EOF(
+    name: "envoy.filters.http.cache"
+    typed_config:
+        "@type": "type.googleapis.com/envoy.extensions.filters.http.cache.v3alpha.CacheConfig"
+        typed_config:
+           "@type": "type.googleapis.com/envoy.source.extensions.filters.http.cache.SimpleHttpCacheConfig"
+    )EOF"};
+  DateFormatter formatter_{"%a, %d %b %Y %H:%M:%S GMT"};
+};
+
+INSTANTIATE_TEST_SUITE_P(Protocols, CacheIntegrationTest,
+                         testing::ValuesIn(HttpProtocolIntegrationTest::getProtocolTestParams()),
+                         HttpProtocolIntegrationTest::protocolTestParamsToString);
+
+TEST_P(CacheIntegrationTest, MissInsertHit) {
+  // Set system time to cause Envoy's cached formatted time to match time on this thread.
+  simTime().setSystemTime(std::chrono::hours(1));
+  initializeFilter(default_config);
+
+  // Include test name and params in URL to make each test's requests unique.
+  const Http::TestRequestHeaderMapImpl request_headers = {
+      {":method", "GET"},
+      {":path", absl::StrCat("/", protocolTestParamsToString({GetParam(), 0}))},
+      {":scheme", "http"},
+      {":authority", "MissInsertHit"}};
+  Http::TestResponseHeaderMapImpl response_headers = {{":status", "200"},
+                                                      {"date", formatter_.now(simTime())},
+                                                      {"cache-control", "public,max-age=3600"},
+                                                      {"content-length", "42"}};
+
+  // Send first request, and get response from upstream.
+  {
+    IntegrationStreamDecoderPtr request = codec_client_->makeHeaderOnlyRequest(request_headers);
```

Comment:
Can you add some integration tests with body, trailers, etc.? This would have caught the stop iteration issue I found.

Reply:
I changed CacheFilter to refuse to cache requests with bodies and/or trailers, because caching GET requests with bodies would turn an ordinary request smuggling vector into cache poisoning, and because GET/HEAD requests with bodies have no meaning. If it turns out to be needed, we may want to add a config knob to allow caching them.

Added unit and integration tests for them.